src/main/java/sirius/db/jdbc/batch/BatchQuery.java

```diff
@@ -94,8 +94,10 @@ protected void tryCommit(boolean cascade) {
             try {
                 Watch w = Watch.start();
                 stmt.executeBatch();
-                stmt.getConnection().commit();
-                avarage.addValues(batchBacklog, w.elapsedMillis());
+                if (!stmt.getConnection().getAutoCommit()) {
```

Comment:
Does ClickHouse ever supported commit of batches? Because this change is actually a performance hit. Committing every single statement separately kills the whole purpose of batching the statements.

Reply:
Answering my own question: No, Clickhouse did not supported batch commits (the commit method was empty, every insert was performed on the fly).
Forcing auto-commit is the only way to ensure previous behavior, or we need to adopt transactions, which apart of ensuring the batch is committed in a block, has no real performance gain according to their benchmarks.

Reply:
100% sure? - CH is very explicit in its documentation to use batch inserts as otherwise the performance is catastrophic..

Reply:
I'd say commits/transactions != batch-inserts (hopefully)...

Reply:
I can only hope that executeBatch does the whole batching under the hood, although I couldn't not figure out yet how its doing it internally at DB-level). In here we basically need to skip sending the commit statement if auto-commit is enabled.